libpod/options.go

```diff
@@ -1878,7 +1878,7 @@ func WithInitCtrType(containerType string) CtrCreateOption {
 			return define.ErrCtrFinalized
 		}
 		// Make sure the type is valid
-		if containerType == define.OneShotInitContainer || containerType == define.AlwaysInitContainer {
+		if containerType == define.OneShotInitContainer || containerType == define.AlwaysInitContainer || containerType == define.IdleInitContainer {
```

Comment:
Consider creating a slice variable with the valid values and check `slices.Contains` as you do in `create.go`